manager/src/main/java/com/redhat/service/bridge/manager/ProcessorServiceImpl.java

```diff
@@ -136,6 +137,51 @@ public Processor createProcessor(String bridgeId, String customerId, ProcessorRe
         return newProcessor;
     }
 
+    @Override
+    @Transactional
+    public Processor updateProcessor(String processorId, String bridgeId, String customerId, ProcessorRequest processorRequest) {
+        // Attempt to load the Bridge. We cannot update a Processor if the Bridge is not available.
+        bridgesService.getReadyBridge(bridgeId, customerId);
+
+        // Extract existing definition
+        Processor existingProcessor = getProcessor(processorId, bridgeId, customerId);
+        ProcessorDefinition existingDefinition = jsonNodeToDefinition(existingProcessor.getDefinition());
+        String existingTransformationTemplate = existingDefinition.getTransformationTemplate();
+        BaseAction existingAction = existingDefinition.getRequestedAction();
+        BaseAction existingResolvedAction = existingDefinition.getResolvedAction();
+
+        // Validate update. Only Filters are supported at present.
+        if (!Objects.equals(existingProcessor.getName(), processorRequest.getName())) {
+            throw new BadRequestException("It is not possible to update the Processor's name.");
+        }
+        if (!Objects.equals(existingTransformationTemplate, processorRequest.getTransformationTemplate())) {
```

Comment:
@manstis Your call, but the removal of this line also gives us https://issues.redhat.com/browse/MGDOBR-515. We could collapse -515 into -13 given that the changes are incredibly similar.

I'll let you decide. Either approach works for me.

cheers,

Rob

Reply:
@robpblake I'll do as a separate PR since I'll need to change the test coverage too.